Doc 1869 update2 insights widget share without login (#1)
tigergraph/insights-docs: modules/intro/pages/applications.adoc

=== Share an application

==== Share via login
* Click on btn:[Share] in the three-dot menu to bring up the Share window.

* Select one of the listed users.

image::share-application.png[]

==== Share via access token

If you want to share your widget with a user who does not have an Insights login, you can send them an access token.

> ==== Creating database account

Reviewer: (1) The heading levels probably need to be adjusted.  "Creating data account" (H4) and subsequent headers are intended to be subsections of "Share via access token", which is also H4. Stepping back, the v2.9.2 page has 5 sections which are H2,H3,H3,H3, and H3. The last one is "Share an application." I think those 5 should actually be all on the top level: H2,H2,H2,H2,H2.  "Share via login" and "Share via access token" should be H3. Then these new subsections can be H4.

Reviewer: (2) Phrasing and consistency of phrasing of section headings:
Most of these sections are one of the steps the user should perform to share an application.  For step-by-step instructions, I think we should use the imperative voice: "Create a database account" / "Configure nginx", etc.  (But " Query permissions required to write/run queries" is not a step; it is informational.)

Author: This was changed in latest commit.

TigerGraph provides two modes for running queries:

* Run a previously **Installed** query
* **Interpret** an ad hoc query

For Insights, **Interpret** query is the default query mode, unless you explicitly convert it to an **Installed** query.
If your widget only needs pre-designed queries, then you may only need **installed** queries.
If you want users to be able to create and run their own queries, then likely they will need to be able to **interpret** queries.

==== Query permissions required to write/run queries.

`WRITE_QUERY` privilege is required to interpret queries or to install queries.
It is not needed to run an installed query.

`READ_DATA` privilege is required for any type of query, and you can grant it at various levels.
(Details are provided in xref:gui:admin-portal:management/user-management.adoc[local privileges])

`READ_SCHEMA` and `READ_QUERY` are also needed for any type of query.

.The minimal required privileges for different types of queries:
[cols='2']
|===
| Query Type | Minimal Required Privileges

| Installed Query | `READ_DATA` ,`READ_SCHEMA`, `READ_QUERY`
| Interpret Query | `READ_DATA` ,`READ_SCHEMA` ,`READ_QUERY`, `WRITE_QUERY`
|===

> ==== (Optional) Config nginx

Reviewer: ```suggestion
==== (Optional) Configure nginx
```
Don't abbreviate.

Author: changed in latest commit.

This is only needed if you want to embed Insights into third party webpages.

* Switch to the Admin Portal tool to configure `nginx`.
* Set the `content-security-policy` parameter as below, substituting in your web page's URL.

[source, console]
Content-Security-Policy: frame-ancestors 'self' https://www.thirdparty.com;

==== Generating access token

Use the account you just created to generate an access token.
The access token is inside a URL that can be embedded on a page as an iframe into a dashboard.

> * Go to `Application > Settings` in the three-dot menu of the specific application.

Reviewer: FYI, I (re)discovered this weekend that Antora has special syntax for buttons, menus, and keyboard keys. 
https://docs.antora.org/antora/latest/asciidoc/ui-macros/
Instead of `Application > Settings`, you could use
menu:Application[ Settings ].

Author: great tip! I utilized it in latest commit.

image::Application2Setting.png[]

* On the application page choose the gear icon for `Settings`.

image::appliactionpage2security2.png[]

* Go to the `Security` tab to create the access token.

image::secuirtyPage.png[]

* Click btn:[Generate new token] button and fill in the username/password details.

> ==== Get the shared link for whole application

Reviewer: Starting from this point until the end, I am confused about what the documentation is trying to tell me.
(1) I (the reader) have just finished "Generating access token". This sounds like it should be the last step. Where do "Get the shared link for **whole application**" and "Obtaining **widget** sharing link" fit in?  Are they additional **required** steps? Are they different **ways to share?** Are they different **things** that I can share?

Author: Added how to share sub section at the end of this section to talk to the user on ways to share the URL Token.

> We also provide a standalone application URL based on the access token.

Reviewer: I think the developer provider some response about **why** the user might want this URL.
As it currently stands, the "why" of these last few sections is not clear.

Author: This is updated in the last commit.

=== Share an application

==== Share via login

> Click on btn:[Share] in the three-dot menu to bring up the Share window.

Reviewer: ```suggestion
Click on btn:[Share] in the three-dot menu to bring up the Share window.
Select one of the listed users.
```

Author: updated in latest commit.

image::share-application.png[]

==== Share via access token

> Access tokens may also be used to share application widgets with other users without a login for insights.

Reviewer: ```suggestion
If you want to share your widget with a user who does not have an Insights login, you can send them an access token.
```

Author: updated in latest commit

===== Creating database account

You will need a separate database account to generate an access token for Insights.
Ensure that you grant minimal privileges to this account.

Please check the xref:tigergraph-server:reference:list-of-privileges.adoc[].

> TigerGraph provides two types of queries: **Installed** and **Interpret**.
> 
> The **Interpret** query is the default query type, unless you explicitly convert it to an **Installed** query.

Reviewer: ```suggestion
TigerGraph provides two modes for running queries:

# Run a previously **Installed** query
# **Interpret** an ad hoc query

For Insights, **Interpret** query is the default query mode, unless you explicitly convert it to an **Installed** query.

If your widget only needs pre-designed queries, then you may only need **installed** queries.  If you want users to be able to create and run their own queries, then likely they will need to be able to **interpret** queries.
```

Author: updated in latest commit

> ==== Write query permission is required to run interpret queries.

Reviewer: ```suggestion
`WRITE_QUERY` privilege is required to interpret queries or to install queries.
It is not needed to run an installed query.
```

Author: updated in latest commit

`READ_DATA` privilege is required for any type of query, and you can grant it at various levels.
(Details are provided in xref:gui:admin-portal:management/user-management.adoc[local privileges])

`READ_SCHEMA` and `READ_QUERY` are also needed for any type of query.

.The minimal required privileges for different types of queries:
[cols='2']
|===
| Query Type | Minimal Required Privileges

| Installed Query | `READ_DATA` ,`READ_SCHEMA`, `READ_QUERY`
| Interpret Query | `READ_DATA` ,`READ_SCHEMA` ,`READ_QUERY`, `WRITE_QUERY`
|===

==== (Optional) Config nginx

> This is only needed if you want to embed insights into third party webpages.

Reviewer: ```suggestion
This is only needed if you want to embed Insights into third party webpages.
```

Author: changed with latest commit.

> Go to admin portal to config `nginx`.

Reviewer: ```suggestion
Switch to the Admin Portal tool to configure `nginx`.
Set the `content-security-policy` parameter as below, substituting in your web page's URL.
```

Author: updated in latest commit

[source, console]
Content-Security-Policy: frame-ancestors 'self' https://www.thirdparty.com;

==== Generating access token

Use the account you just created to generate an access token.

Go to `Application > Settings > Security` to create the access token.

> At the security page, we also provide a standalone application URL based on the access token.

Reviewer: As a user, I don't understand what to do. 

1. Is this the Application page of the Admin Portal or Insights? (I _think_ we want the user to switch back from Admin Portal to Insights. That needs to be said explicitly.)
2. Once I am at the Security page, what do I do?
3. Why are you mentioning "also provide a ... URL"? Why do I care? What can/should I do with this URL?

Author: Asked the developer in the ticket. updated in latest commit based on their answers.
